### DISCUSSION TITLE
Solid cell/nodes in Amrex without using Embedded Boundary functions

### DISCUSSION THREAD

**Discussion by LordOfBunnies** [AMReX-Codes/amrex#2334] [Q&A]
I'm currently writing a Lattice Boltzmann Method flow solver in Fortran 95.  The code is node based, making it a little more fun.

I'm wondering about a way to specify solid geometry without using the Embedded Boundary system already in Amrex.  My code is able to determine whether a node is within a solid.  It takes .stl files or .cgns files with tessellated surfaces.  However, Amrex does not know if the node is a fluid or solid.  I figure this may end up causing problems when using things like average_down, fill_patch, etc. when it's near the wall.

I've been reading through the embedded boundary system, but it doesn't look like it can handle generic geometry.  I could certainly be wrong. 

Is there a way to inform Amrex which nodes are solid/fluid/etc.? Can the embedded boundary system be used with generic geometry?  Is there a way to work around the system and directly tell the EB system which ones are which?

**Comment by WeiqunZhang** [comment]
What EB information do you need? If it's just whether a node is inside or outside a solid body, you can save that information in an `iMultiFab`.

**Reply by LordOfBunnies** [reply]
I'm already saving the information in an iMultiFab, but I didn't see a way to let Amrex know that information.  It may be because I'm using Fortran and the EB stuff is C++.  I'm worried about using things like fill_patch and average_down as they might pick up bad information from inside the solid.

Right now, I'm designating things as fluid or solid, and that determines how the equations interact.  I don't think Amrex knows that those nodes are fluid/solid/etc.  Is it using EBCellFlagFab?  Can that work for nodes as well?  I've already got the system doing interpolative bounce-back, so I don't need cut cell information, just whether it's a valid node or not.

**Reply by WeiqunZhang** [reply]
Right.  AMReX won't know about the EB information.  But you might be able handle it yourself.
